charmhub_lp_tools/main.py

```diff
@@ -567,6 +577,154 @@ def parse_args(config_from_file: FileConfig) -> argparse.Namespace:
     )
     ch_report_commands.set_defaults(func=ch_report_main)
 
+    # clean-channel
+    clean_channel_command = subparser.add_parser(
+        'clean-channel',
+        help=('Clean a channel (track/risk) by specifying the base(s) to '
+              'keep. This will find the relevant revisions for those bases, '
+              'then close the channel, and finally re-release the revisions '
+              'back to that channel. NOTE: this is quite dangerous - use '
+              'sparingly!.'),
+    )
+    clean_channel_command.add_argument(
+        '--i-really-mean-it',
+        dest='confirmed',
+        action='store_true',
+        default=False,
+        help=('This flag must be supplied to indicate that the operation '
+              'should really commit the changes.')
+    )
+    clean_channel_command.add_argument(
+        '-s', '--source', dest='src_channel',
+        metavar='CHANNEL', required=True,
+        help='The channel to clean.',
+    )
+    clean_channel_command.add_argument(
+        '--base',
+        dest='bases',
+        action='append',
+        metavar='BASE',
+        required=True,
+        type=str,
+        help=('Select charm(s) that run on the base (e.g. 20.04, 22.04). '
+              'Can be used multiple times.')
+    )
+    clean_channel_command.add_argument(
+        '--ignore-arch',
+        dest='ignore_arches',
+        action='append',
+        metavar='IGNORE_ARCH',
+        type=str,
+        help='Architecture to ignore/not copy; repeat for multiple arches'
+    )
+    clean_channel_command.add_argument(
+        '--retries', metavar='N',
+        dest='retries',
+        type=int,
+        default=3,
+        help='Retry calls when charmhub issues a 504 error',
+    )
+    clean_channel_command.set_defaults(func=clean_channel)
+
+    # promote
+    change_risk_command = subparser.add_parser(
```

Comment:
I originally had it as "promote-charm" but then I realised that it could go in either direction.  (which sounds odd, but is, indeed, possible).  Maybe "copy-using-bases" is a better description of what is going on?  It is basically a simplified version "copy-channel" where most of the arguments are derived from the charm and the config.